distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerProcessBuilder.java

```diff
@@ -108,6 +108,7 @@ private List<String> getJvmArgs() {
             esHome.resolve("lib").toString(),
             // Special circumstances require some modules (not depended on by the main server module) to be explicitly added:
             "--add-modules=jdk.net", // needed to reflectively set extended socket options
+            "--add-modules=jdk.crypto.ec", // this module is unnecessarily required by com.nimbus.jose.jwt
```

Comment:
well, this awkward... https://bugs.openjdk.org/browse/JDK-8312267
might want to run the integration tests with Java22 and ensure no scary warnings are emitted.
also, we should run the FIPS tests in CI (IIRC just need to add the label) to ensure no issues when we change the default crypto provider(s).

Reply:
I think this will cause problems later. If the module is unnecessary, can nimbus be convinced to omit it from its module-info, or can we vender the library and adjust the module-info ourselves?

Reply:
> or can we vender the library and adjust the module-info ourselves?

This should be an absolute last resort. We have enough forks, special packaging, etc of various libraries.

Reply:
If nimbus does not remove the dependency then I think it's our only option to avoid warnings from the jdk (and then breakage when the module is actually removed).

Reply:
This turned out not to be necessary; this was for a different version of nimbus-jose-jwt that ended up not being usable for other reasons.